staging/src/k8s.io/apiserver/pkg/storageversion/manager.go

```diff
@@ -46,13 +48,15 @@ type ResourceInfo struct {
 type Manager interface {
 	// AddResourceInfo adds ResourceInfo to the manager.
 	AddResourceInfo(resources ...*ResourceInfo)
-	// RemoveResourceInfo removes ResourceInfo from the manager.
-	RemoveResourceInfo(r *ResourceInfo)
+	// MarkCompletedResourceInfo marks updated ResourceInfo as completed.
+	MarkCompletedResourceInfo(r *ResourceInfo)
+	// RecordErrorResourceInfo records latest error updating ResourceInfo.
+	RecordErrorResourceInfo(r *ResourceInfo, err error)
```

Comment:
Can we unify the two method names? Maybe `RecordResourceInfoSuccess` and `RecordResourceInfoFailure`.